lib/posix/spawn/child.rb

```diff
@@ -223,6 +248,10 @@ def read_and_write(input, stdin, stdout, stderr, timeout=nil, max=nil)
         slice_method = input.respond_to?(:byteslice) ? :byteslice : :slice
         t = timeout
 
+        streams = {stdout => @stdout_stream, stderr => @stderr_stream}
+
+        bytes_seen = 0
+        chunk_buffer = ""
```

Comment:
This buffer is reused by `readpartial` below. Internally, so far as I can tell, it will be resized to `BUFSIZE` on the first call and the underlying buffer reused from then on out.